### PR TITLE
Feature: IndexTable checkbox hover state

### PR DESCRIPTION
**Pull request by ninariccimarie**

### CODE REVIEW

File: src/@next/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -76,10 +80,14 @@ export const Checkbox = React.forwardRef<HTMLDivElement, CheckboxProps>(
         aria-disabled={disabled}
         data-error={hasError}
         data-spacing={isPadded}
+        data-in-table={inTable}
       >
-        <StyledRow>
+        <StyledRow data-in-table={inTable}>
```
Comment (marthayaputra) [inline src/@next/Checkbox/Checkbox.tsx]: I'm not sure if this sustainable when we have another need to put checkbox in another control, wouldn't it better to put it on IndexTable's checkbox?

File: src/@next/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -40,6 +42,8 @@ export const Checkbox = React.forwardRef<HTMLDivElement, CheckboxProps>(
       hasError,
       helpText,
       isPadded = true,
+      isHovered = false,
```
Comment (marthayaputra) [inline src/@next/Checkbox/Checkbox.tsx]: Why do we need this? 🤔

**Review by marthayaputra — CHANGES_REQUESTED** [review]

File: src/@next/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (ninariccimarie) [reply src/@next/Checkbox/Checkbox.tsx]: you're right, I should move this to IndexTable instead

File: src/@next/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (ninariccimarie) [reply src/@next/Checkbox/Checkbox.tsx]: it's to trigger hover when Checkbox table cell is hovered. Basically, it widens the area of point to trigger Checkbox hover. it's one of the requirements for the hover state

Comment (marthayaputra) [reply src/@next/Checkbox/Checkbox.tsx]: Oh ic, can't we rely on native html hover behavior?

Comment (ninariccimarie) [reply src/@next/Checkbox/Checkbox.tsx]: we can't trigger checkbox hover with just html when the table cell is hovered though?

**Review by marthayaputra — APPROVED** [review]
LGTM

**Pull request merged**